File: src/main/org/nlogo/prim/_or.java
package org.nlogo.prim ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _or
	extends Reporter
	implements org.nlogo.nvm.Pure, org.nlogo.nvm.CustomGenerated
{
	@Override
	public Syntax syntax()
	{
		return Syntax.reporterSyntax( Syntax.TYPE_BOOLEAN ,
									  new int[] { Syntax.TYPE_BOOLEAN } ,
									  Syntax.TYPE_BOOLEAN ,
									  Syntax.NORMAL_PRECEDENCE - 6 ) ;
	}
	@Override public Object report( final Context context ) throws LogoException
	{
		return argEvalBooleanValue( context , 0 )
			? Boolean.TRUE
			: argEvalBoolean( context , 1 ) ;
	}
}

File: src/main/jvm/org/nlogo/prim/_or.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter, Pure, CustomGenerated }

class _or extends Reporter with Pure with CustomGenerated {
  override def syntax =
    Syntax.reporterSyntax(
      Syntax.BooleanType,
      Array(Syntax.BooleanType),
      Syntax.BooleanType,
      Syntax.NormalPrecedence - 6)
  override def report(context: Context) =
    if (argEvalBooleanValue(context, 0))
      java.lang.Boolean.TRUE
    else
      argEvalBoolean(context, 1)
}
